modules/control/common/mrac_controller.cc

```diff
@@ -247,27 +232,30 @@ void MracController::BuildReferenceModel() {
     matrix_a_reference_(1, 1) = -2 * zeta_reference_ * wn_reference_;
     matrix_b_reference_(1, 0) = wn_reference_ * wn_reference_;
   }
+  return Status::OK();
 }
 
-void MracController::BuildAdaptionModel() {
-  adaption_model_enabled_ = true;
+Status MracController::BuildAdaptionModel() {
   if (model_order_ > 2) {
-    AWARN << "Adaption model order beyond the designed range, "
-             "model_order: "
-          << model_order_;
-    adaption_model_enabled_ = false;
-    return;
+    const auto error_msg =
+        StrCat("mrac controller error: adaption model order ", model_order_,
+               " is beyond the designed range");
+    AERROR << error_msg;
+    return Status(ErrorCode::CONTROL_INIT_ERROR, error_msg);
   }
   if (model_order_ == 1) {
     matrix_b_adaption_(0, 0) = 1.0;
   } else if (model_order_ == 2) {
     matrix_b_adaption_(1, 0) = wn_reference_ * wn_reference_;
   }
   if (!CheckLyapunovPD(matrix_a_reference_, matrix_p_adaption_)) {
-    AWARN << "Solution of the algebraic Lyapunov equation is not symmetric "
-             "positive definite";
-    adaption_model_enabled_ = false;
+    const auto error_msg =
+        "Solution of the algebraic Lyapunov equation is not symmetric positive "
+        "definite";
```

Comment:
The `auto` here becomes `char*`. Please use `std::string`.

Reply:
Fixed per comment. Thanks.